src/dinglehopper/ocrd_cli.py

```diff
@@ -1,83 +1,73 @@
-        kwargs["version"] = OCRD_TOOL["version"]
-        super(OcrdDinglehopperEvaluate, self).__init__(*args, **kwargs)
 
-    def process(self):
-        assert_file_grp_cardinality(self.input_file_grp, 2, "GT and OCR")
-        assert_file_grp_cardinality(self.output_file_grp, 1)
+    @cached_property
+    def executable(self):
+        return 'ocrd-dinglehopper'
 
-        log = getLogger("processor.OcrdDinglehopperEvaluate")
+    def process_page_file(self, *input_files: Optional[OcrdFileType]) -> None:
 
+        assert self.parameter
         metrics = self.parameter["metrics"]
         textequiv_level = self.parameter["textequiv_level"]
-        gt_grp, ocr_grp = self.input_file_grp.split(",")
-
-        input_file_tuples = self.zip_input_files(on_error="abort")
-        for n, (gt_file, ocr_file) in enumerate(input_file_tuples):
-            if not gt_file or not ocr_file:
-                # file/page was not found in this group
-                continue
-            gt_file = self.workspace.download_file(gt_file)
-            ocr_file = self.workspace.download_file(ocr_file)
-            page_id = gt_file.pageId
 
-            log.info("INPUT FILES %i / %s↔ %s", n, gt_file, ocr_file)
-
-            file_id = make_file_id(ocr_file, self.output_file_grp)
-            report_prefix = os.path.join(self.output_file_grp, file_id)
-
-            # Process the files
-            try:
-                os.mkdir(self.output_file_grp)
-            except FileExistsError:
-                pass
-            cli_process(
-                gt_file.local_filename,
-                ocr_file.local_filename,
-                report_prefix,
-                metrics=metrics,
-                textequiv_level=textequiv_level,
+        try:
+            gt_file, ocr_file = input_files
+            assert gt_file, 'missing GT file'
+            assert ocr_file, 'missing OCR file'
+            assert gt_file.local_filename
+            assert ocr_file.local_filename
+        except (ValueError, AssertionError) as err:
+            self.logger.warning(f'Missing either GT file, OCR file or both: {err}') # TODO how to log which page?
+            return
+
+        page_id = gt_file.pageId
+
+        file_id = make_file_id(ocr_file, self.output_file_grp)
+        report_prefix = os.path.join(self.output_file_grp, file_id)
+
+        # Process the files
+        try:
+            os.mkdir(self.output_file_grp)
+        except FileExistsError:
+            pass
+        cli_process(
+            gt_file.local_filename,
+            ocr_file.local_filename,
+            report_prefix,
+            metrics=metrics,
+            textequiv_level=textequiv_level,
+        )
+
+        # Add reports to the workspace
+        for report_suffix, mimetype in [
+            [".html", "text/html"],
+            [".json", "application/json"],
+        ]:
+            self.workspace.add_file(
+                file_id=file_id + report_suffix,
```

Comment:
v3 at this points requires us to deal with existing files of that ID:

```suggestion
            output_file_id = file_id + report_suffix
            output_file = next(self.workspace.mets.find_files(ID=output_file_id), None)
            if output_file and config.OCRD_EXISTING_OUTPUT != 'OVERWRITE':
                raise FileExistsError(f"A file with ID=={output_file_id} already exists {output_file} and neither force nor ignore are set")
            self.workspace.add_file(
               file_id=output_file_id,
```